QuizApp/ViewControllers/LoginViewController.swift

```diff
@@ -257,33 +257,38 @@ class LoginViewController: UIViewController {
+        backgroundQueue.sync {
+            success = networkService.login(username: username, password: password)
+        }
+        
+        guard let success = success else {  return }
+        switch success {
+        case LoginStatus.success:
+            print("E-mail:  \(username)")
+            print("Password: \(password)")
+            
+            let quizzesViewController = QuizzesViewController()
+            quizzesViewController.tabBarItem = UITabBarItem(title: "Quiz", image:  UIImage(systemName: "stopwatch"), selectedImage: UIImage(systemName: "stopwatch.fill"))
+            let settingsViewController = SettingsViewController()
+            settingsViewController.tabBarItem = UITabBarItem(title: "Settings", image:  UIImage(systemName: "gearshape"), selectedImage: UIImage(systemName: "gearshape.fill"))
+            
+            let tabBarController = UITabBarController()
+            tabBarController.viewControllers = [quizzesViewController, settingsViewController]
+            self.navigationController?.setViewControllers([tabBarController], animated: true)
+            
+        case LoginStatus.error(_,_):
+            print("Error: Wrong password or username")
+            hiddenErrorLabel.isHidden = false
+            hiddenErrorLabel.text = "Error: Wrong password or username"
         
-        switch loginStatus {
-            case LoginStatus.success:
-                print("E-mail: ", email!)
-                print("Password: ", password!)
-                
-                let quizzesViewController = QuizzesViewController()
-                quizzesViewController.tabBarItem = UITabBarItem(title: "Quiz", image:  UIImage(systemName: "stopwatch"), selectedImage: UIImage(systemName: "stopwatch.fill"))
-                let settingsViewController = SettingsViewController()
-                settingsViewController.tabBarItem = UITabBarItem(title: "Settings", image:  UIImage(systemName: "gearshape"), selectedImage: UIImage(systemName: "gearshape.fill"))
-                
-                let tabBarController = UITabBarController()
-                tabBarController.viewControllers = [quizzesViewController, settingsViewController]
-                
-                let newNavigationController = UINavigationController(rootViewController: tabBarController)
-                newNavigationController.modalPresentationStyle = .overFullScreen
-                newNavigationController.navigationBar.barTintColor = Color().colorBackground
-                self.navigationController?.present(newNavigationController, animated: true, completion: nil)
-                
-            case LoginStatus.error(let code, let text):
-                print("Error: \(text) (\(code))")
-                hiddenErrorLabel.isHidden = false
-                hiddenErrorLabel.text = "Error: \(text) (\(code))"
-                
+        case LoginStatus.noInternetConnection:
```

Comment:
Like I mentioned, noInternetConnection isn’t a login status. You should handle it as an error instead.